pkg/client/restclient.go

```diff
@@ -35,10 +35,13 @@ import (
 // Most consumers should use client.New() to get a Kubernetes API client.
 type RESTClient struct {
 	baseURL *url.URL
+	// A string identifying the version of the API this client is expected to use.
+	apiVersion string
 
-	// namespaceInPath controls if URLs should encode the namespace as path param instead of query param
-	// needed for backward compatibility
-	namespaceInPath bool
+	// LegacyBehavior controls if URLs should encode the namespace as a query param,
+	// and if resource case is preserved for supporting older API conventions of
+	// Kubernetes.  Newer clients should leave this false.
+	LegacyBehavior bool
```

Comment:
Debated, figured someone with a RESTClient might want to know whether it's using old rules or not.  I had all optional fields as public, and required fields as hidden... but then Codec is public so people can access it.  I guess I could move that to a method...

----- Original Message -----
> >  
> > -	// namespaceInPath controls if URLs should encode the namespace as path
> > param instead of query param
> > -	// needed for backward compatibility
> > -	namespaceInPath bool
> > +	// LegacyBehavior controls if URLs should encode the namespace as a query
> > param,
> > +	// and if resource case is preserved for supporting older API conventions
> > of
> > +	// Kubernetes.  Newer clients should leave this false.
> > +	LegacyBehavior bool
> 
> Should this be exported?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/3269/files#r22624128